### DISCUSSION TITLE
can Kodi and my PMS be on the same device?

### DISCUSSION THREAD

**Discussion by TimFH** [croneter/PlexKodiConnect#2116] [Q&A] [answered]
I have a Vero V running OSMC/Kodi, but I have not yet set up a Plex Media Server. Given that I'd like to keep everything in one place, I'm inclined to install my PMS server on the Vero V as well (following the instructions here: https://discourse.osmc.tv/t/howto-install-plex-media-server-and-add-on-on-rasbperry-pi/78576). Before doing so, however, I'd like to know what are the pros and cons of hosting both Kodi and PMS on the same device (bandwidth, at least, should not be an issue). Thanks in advance, TFH

**Comment by croneter** [comment, marked as answer]
Yes that works just fine. It's even very easy to use direct paths then - just activate direct paths, no mapping necessary (check the wiki here for more info)

**Reply by TimFH** [reply]
TY! I've now installed PMS on my Vero V running OSMC/Kodi Omega, and chosen direct paths (with UNC & other paths replaced by SMB). I've tested the PMS by playing a music album from my laptop using the Plex browser interface, and that works fine.

On the Vero V, I see both a PKC addon under Video and under Music, but hitting "Plex Hub" or anything else in the music addon (I haven't added any videos yet) just says "no info available". So I tried the Plex addon, and got a "Plex error", which the python stack trace in the Kodi log says is "cannot import name 'VerifiedHTTPSConnection' from 'urllib3.connectionpool".

Is something wrong with my PKC setup, or if not, how do I play anything from within Kodi?

**Comment by TimFH** [comment]
After rebooting a couple of times Kodi PKC sync'ed my Kodi music library with Plex OK, but the Kodi Plex addon still crashes. In addition to the above-mentioned python stack trace message, the log also contains a lot of warnings about "smb://media/MYDISK/DIR/FILE does not exist -- skipping scan" followed by errors "Unable to lookup host: media."

And now when I open the PKC Music addon, I do see my library, but when I attempt to play something in it, it fails to do so without any error message at all, and skips to the next song, which also fails, until it's gone through the whole album, and then it stops. This time the Kodi log is full of messages about being unable to open files via the "smb:" interface (as above), as well as errors "Unable to lookup host: 'folder.jpg'" followed by "NFS: Failed to mount nfs share: ... invalid address:, can not resolv into PIv4/v4 structure". There are also warnings such as "Setting most music properties through ListItem.setInfo() is depreicated" and "PAPlayer::QueueNextFileEx - Failed to create the decoder" followed by error "CAudioDecoder: Uanble to Init Codec while loading ...".

In short, it looks like I'm going to need some help getting PKC to work.

**Reply by croneter** [reply]
As stated above: do NOT replace any paths in the PKC settings if Kodi and the Plex Media Server are on the same machine. Then you need to reset the Kodi database, do that in the PKC settings under Advanced (in red)
